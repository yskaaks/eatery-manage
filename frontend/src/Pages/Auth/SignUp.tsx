// SignUp.tsx
import React, { useState } from "react";
import { useForm } from "react-hook-form";
import { Autocomplete, useJsApiLoader } from "@react-google-maps/api";
import "../../styles/SignUp.css";
import { useAuth } from "../../hooks/useAuth";
import { useNavigate } from "react-router-dom";
import { RegisterFormInputs } from "../../interface";

const libraries: (
  | "places"
  | "drawing"
  | "geometry"
  | "localContext"
  | "visualization"
)[] = ["places"];

const SignUp: React.FC = () => {
  const { isLoaded } = useJsApiLoader({
    id: "google-map-script",
    googleMapsApiKey: import.meta.env.VITE_GOOGLE_MAPS_API_KEY,
    libraries: libraries,
  });

  // Create a ref to hold the autocomplete instance
  const autocompleteRef = React.useRef<google.maps.places.Autocomplete | null>(
    null
  );

  const onLoad = (autocomplete: google.maps.places.Autocomplete) => {
    autocompleteRef.current = autocomplete;

    // Customize the Autocomplete instance here
    autocomplete.setOptions({
      types: ["geocode"], // Types to search for, in this case, geocode to get address suggestions
      componentRestrictions: { country: "au" }, // Restrict results to Australia
      fields: ["formatted_address", "geometry"], // Only retrieve formatted_address and geometry data
    });
  };

  const handlePlaceChanged = () => {
    const autocomplete = autocompleteRef.current;
    if (autocomplete) {
      const place = autocomplete.getPlace();
      if (place) {
        setValue("location", place.formatted_address);
        setValue("latitude", place.geometry.location.lat());
        setValue("longitude", place.geometry.location.lng());
      }
    }
  };

  const { register, handleSubmit, setValue } = useForm<RegisterFormInputs>();
  const [role, setRole] = useState<string>("");
  const { register: registerUser } = useAuth();
  // success or failure to signUp/signIn message shown to client
  const [message, setMessage] = useState("");
  const navigate = useNavigate();

  const validateForm = (data: RegisterFormInputs) => {
    const { name, email, password, location, latitude, longitude } = data;
    if (!name) return "Please enter your name";
    if (!email) return "Please enter your email";
    if (!password) return "Please enter your password";
    if (!role) return "Please select a role";
    if (!location && role === "eatery") return "Please enter your location";
    if ((!latitude || !longitude) && role === "eatery") return "Please provide latitude and longitude";
    return null;
  };

  const onSubmit = async (data: RegisterFormInputs) => {
    // Update this to include userType from state, not from form
    const { name, email, password, location, latitude, longitude } = data;

    const validationError = validateForm(data);
    if (validationError) {
      setMessage(validationError);
      return;
    }

    try {
      const success = await registerUser(
        email,
        password,
        name,
        role,
        location,
        latitude,
        longitude
      );

      if (success) {
        setMessage("success");
        const userRole = localStorage.getItem("role");
        const id = localStorage.getItem("id");
        if (userRole === "eatery") {
          navigate(`/restaurant/${id}`);
        } else {
          navigate("/auth/cuisine-form");
        }
      } else {
        setMessage("email is already registered");
      }
    } catch {
      setMessage("email is already registered");
    }
  };  

  return (
    <div className="signup-container">
      <h2 className="signup-title">
        Create an account or&nbsp;
        <div onClick={() => navigate("/auth/login")} className="title-link">
          log in
        </div>
      </h2>
      <p className="signup-sub-title">Create a new account below or log in</p>
      <form onSubmit={handleSubmit(onSubmit)} className="signup-form">
        <input
          {...register("name")}
          placeholder="Name"
          className="input-field"
        />
        <input
          {...register("email")}
          placeholder="Email"
          type="email"
          className="input-field"
        />
        <input
          {...register("password")}
          placeholder="Password"
          type="password"
          className="input-field"
        />
        <div className="user-type-select">
          <button
            type="button"
            onClick={() => setRole("customer")}
            className={role === "customer" ? "selected" : ""}
          >
            I'm a Customer
          </button>
          <button
            type="button"
            onClick={() => setRole("eatery")}
            className={role === "eatery" ? "selected" : ""}
          >
            I'm a Restaurant Owner
          </button>
        </div>
        {role === "eatery" && isLoaded && (
          <Autocomplete
            onLoad={onLoad}
            onPlaceChanged={handlePlaceChanged}
            className="address-autocomplete"
          >
            <input
              {...register("location")}
              placeholder="Restaurant Address"
              className="input-field"
            />
          </Autocomplete>
        )}

        <button type="submit" className="submit-button">
          Sign Up
        </button>
      </form>
      {message && <p>{message}</p>}
    </div>
  );
};

export default SignUp;