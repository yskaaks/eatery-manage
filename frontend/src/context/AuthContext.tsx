import React, { createContext, useState, useCallback } from "react";
import axios from "axios";
import { AuthContextType, Props, User } from "../interface";

export const AuthContext = createContext<AuthContextType | undefined>(
  undefined
);

export const AuthProvider: React.FC<Props> = ({ children }) => {
  const checkToken = localStorage.getItem("token");
  const [token, setToken] = useState<string | null>(
    checkToken ? checkToken : null
  );
  const [user, setUser] = useState<User | null>(null);
  const api = axios.create({
    baseURL: "http://127.0.0.1:5000",
  });

  const login = useCallback(
    async (email: string, password: string, role: string) => {
      try {
        const response = await api.post("/api/auth/login", {
          email,
          password,
          role,
        });
        const { token, role: userRole, id } = response.data;
        localStorage.setItem("token", token);
        localStorage.setItem("role", userRole);
        localStorage.setItem("id", id);
        setToken(token);
        return true;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    []
  );

  const googleLogin = useCallback(async (code: string) => {
    try {
      const response = await api.post("/api/auth/validate-google-token", {
        code,
      });
      console.log(response);
      const { token } = response.data;
      localStorage.setItem("token", token);
      setToken(token);
      return true;
    } catch (error) {
      console.error(error);
      return false;
    }
  }, []);

  const register = useCallback(
    async (
      email: string,
      password: string,
      name: string,
      role: string,
      location?: string,
      latitude?: number,
      longitude?: number
    ) => {
      try {
        const response = await api.post("/api/auth/register", {
          email,
          password,
          name,
          role,
          location,
          latitude,
          longitude
        });
        console.log(response);
        const { token, role: userRole, id } = response.data;
        localStorage.setItem("token", token);
        localStorage.setItem("role", userRole);
        localStorage.setItem("id", id);
        setToken(token);
        return true;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    []
  );

  const logout = useCallback(async () => {
    try {
      await api.get("api/auth/logout", {
        headers: { Authorization: `Bearer ${token}` },
      });
      localStorage.clear();
      setToken(null);
      return true;
    } catch (error) {
      console.error(error);
      return false;
    }
  }, [token]);

  const passwordResetRequest = useCallback(
    async (email: string, role: string) => {
      try {
        await api.post("/api/auth/passwordreset/request", { email, role }); // include role here
        return true;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    []
  );

  const updateProfile = useCallback(
    async (name: string, email: string) => {
      try {
        const response = await api.post(
          "api/customer/edit-profile",
          { email, name },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        return response.data;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    [token]
  );

  const updateEateryUser = useCallback(
    async (restaurant_name: string, email: string) => {
      try {
        const response = await api.put(
          "api/eatery/edit-profile",
          { email, restaurant_name },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        return response.data;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    [token]
  );

  const passwordReset = useCallback(
    async (resetToken: string, newPassword: string) => {
      try {
        await api.post("/api/auth/password/reset", { resetToken, newPassword });
        return true;
      } catch (error) {
        console.error(error);
        return false;
      }
    },
    []
  );

  const isAuthenticated = () => Boolean(token);

  const fetchUser = useCallback(async () => {
    try {
      const response = await api.get("api/auth/whoami", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      setUser(response.data);
    } catch (error) {
      console.error(error);
    }
  }, [token]);

  const getAllReviews = useCallback(
    async (eateryId: string) => {
      try {
        const response = await api.post(
          "api/get_all_reviews",
          { eatery_id: eateryId },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        return response.data.reviews;
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const getUserById = useCallback(
    async (id: string) => {
      try {
        const response = await api.get(`/api/user/${id}`);
        return response.data;
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const updatePassword = useCallback(
    async (currentPassword: string, newPassword: string) => {
      try {
        const response = await api.post(
          "api/auth/password/update",
          { current_password: currentPassword, new_password: newPassword },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );

        return response.data.success;
      } catch (error) {
        console.error(error);
      }
    },
    [token, api]
  );

  return (
    <AuthContext.Provider
      value={{
        updateProfile,
        updateEateryUser,
        googleLogin,
        getUserById,
        getAllReviews,
        fetchUser,
        isAuthenticated,
        login,
        logout,
        register,
        passwordResetRequest,
        passwordReset,
        setUser,
        user,
        token,
        updatePassword,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

export default AuthProvider;