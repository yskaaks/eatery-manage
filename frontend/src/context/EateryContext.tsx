// EateryContext.tsx

import axios from "axios";
import React, { createContext, useState, useCallback } from "react";
import {
  Eatery,
  EateryContextProps,
  Images,
  Props,
  Review,
  Cuisine,
  CreateOpeningHours,
  UpdateLoyaltyPoints,
} from "../interface";
import { useAuth } from "../hooks/useAuth";

export const EateryContext = createContext<EateryContextProps | undefined>(
  undefined
);

export const EateryProvider: React.FC<Props> = ({ children }) => {
  const [eateries, setEateries] = useState<Array<Eatery>>([]);
  const [recommendedEateries, setRecEateries] = useState({});
  const [eatery, setEatery] = useState<Eatery | null>(null);
  const [review, setReview] = useState<Review | null>(null);
  const [allReviews, setallReviews] = useState<Array<Review>>([]);
  const [allCuisines, setAllCuisines] = useState<Array<Cuisine>>([]);
  const [eateryImages, setEateryImages] = useState<Images | null>(null);
  const { token } = useAuth();

  const api = axios.create({
    baseURL: "http://127.0.0.1:5000",
  });

  const addImage = useCallback(
    async (imageFile: File) => {
      try {
        const formData = new FormData();
        formData.append("file", imageFile);
        const config = {
          headers: {
            "content-type": "multipart/form-data",
            Authorization: `Bearer ${token}`,
          },
        };
        const response = await api.post("/api/add_image", formData, config);
        return response.data.success; // return the success status
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const deleteImage = useCallback(
    async (imageId: string) => {
      try {
        const config = {
          headers: {
            Authorization: `Bearer ${token}`,
          },
          data: { image_id: imageId },
        };
        const response = await api.delete("/api/delete_image", config);
        return response.data.success; // return the success status
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const getEateryImage = useCallback(async (imageId: string) => {
    try {
      const response = await api.get(`/api/get_image/${imageId}`, {
        responseType: "blob",
      });
      const blob = new Blob([response.data], { type: "image/jpeg" });
      const imageUrl = window.URL.createObjectURL(blob);
      return imageUrl;
    } catch (error) {
      console.error(error);
    }
  }, []);

  const fetchEateries = useCallback(async () => {
    try {
      const response = await api.get("/api/eatery", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      setEateries(response.data);
    } catch (error) {
      console.error(error);
    }
  }, [token]);

  const fetchEatery = useCallback(
    async (id: string) => {
      try {
        const response = await api.get(`/api/eatery/${id}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        setEatery(response.data);
        return response.data;
      } catch (error) {
        console.error(error);
        return null;
      }
    },
    [token]
  );

  const fetchEateryImages = useCallback(
    async (eateryId: string) => {
      try {
        const response = await api.post(
          "/api/get_images",
          { eatery_id: eateryId },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        setEateryImages(response.data);
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const getReview = useCallback(
    async (eateryId: string) => {
      try {
        const response = await api.post(
          "/api/get_review",
          { eatery_id: eateryId },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        setReview(response.data);
        return response.data;
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const getAllReviews = useCallback(
    async (eateryId: string) => {
      try {
        const response = await api.post(
          "/api/get_all_reviews",
          { eatery_id: eateryId },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        setallReviews(response.data.reviews);
        return response.data.reviews;
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const addReview = useCallback(
    async (eateryId: string, rating: string, reviewText: string) => {
      try {
        const response = await api.post(
          "/api/add_review",
          {
            eatery_id: eateryId,
            rating,
            review_text: reviewText,
          },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        return response.data; // return the success status
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const deleteReview = useCallback(
    async (reviewId: string) => {
      try {
        const response = await api.delete(`/api/delete_review/${reviewId}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        return response.data.success; // return the success status
      } catch (error) {
        console.error(error + " ASSS");
      }
    },
    [token]
  );

  const getAllCuisines = useCallback(async () => {
    try {
      const response = await api.get("/api/cuisines", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      setAllCuisines(response.data);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }, [token]);

  // Add or Update Menu Cuisines
  const addMenuCuisines = useCallback(
    async (cuisineIds: number[]) => {
      try {
        const response = await api.post(
          "/api/cuisines",
          { cuisineIds },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        return response.data; // return the success status
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  // Update Eatery Open Hours
  const addOpenHours = useCallback(
    async (formData: CreateOpeningHours[]) => {
      try {
        const response = await api.post(`/api/eatery/opening_hours`, formData, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });

        return response.data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },
    [token]
  );
  // Add or Deduct Customer loyality points from scanned screen
  const updateLoyaltyPoints = useCallback(
    async (formData: UpdateLoyaltyPoints) => {
      try {
        const response = await api.post(`/api/loyalty/points`, formData, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });

        return response.data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },
    [token]
  );

  // Add Scaaned Customer to loyalty program
  const addCustomerToLoyalty = useCallback(async (eateryId: string, customerId: string) => {
    try {
      await api.get(`/api/loyalty/program/${eateryId}/${customerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      
      return true;
    } catch (error) {
      console.error(error);
      return false
    }
  }, [token, api]);

  // Get Recommended Eateries
  const fetchRecommendedEateries = useCallback(
    async (lat: number, lon: number) => {
      try {
        const response = await api.get(`/api/get_eatery_preferences?lat=${lat}&lon=${lon}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        setRecEateries(response.data);
        return response.data;
      } catch (error) {
        console.error(error);
        return null;
      }
    },
    [token, api]
  );
  return (
    <EateryContext.Provider
      value={{
        fetchEateryImages,
        fetchEateries,
        eateries,
        fetchEatery,
        eatery,
        eateryImages,
        token,
        review,
        allReviews,
        getReview,
        getAllReviews,
        addReview,
        deleteReview,
        addImage,
        getEateryImage,
        deleteImage,
        getAllCuisines,
        allCuisines,
        addMenuCuisines,
        addOpenHours,
        updateLoyaltyPoints,
        addCustomerToLoyalty,
        recommendedEateries,
        fetchRecommendedEateries,
      }}
    >
      {children}
    </EateryContext.Provider>
  );
};