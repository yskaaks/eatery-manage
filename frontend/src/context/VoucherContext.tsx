// VoucherContext.tsx
import React, { createContext, useState, useCallback } from "react";
import axios from "axios";
import { Voucher, VoucherContextProps, Props, AddVoucher } from "../interface"; // Define these types according to your needs
import { useAuth } from "../hooks/useAuth";

export const VoucherContext = createContext<VoucherContextProps | undefined>(
  undefined
);

interface ErrorResponse {
  response?: {
    data?: {
      vouchers?: string;
    };
  };
}

export const VoucherProvider: React.FC<Props> = ({ children }) => {
  const [customerVouchers, setCustomerVouchers] = useState<Array<Voucher>>([]);
  const [eateryVouchers, setEateryVouchers] = useState<Array<Voucher>>([]);
  const { token } = useAuth();

  const api = axios.create({
    baseURL: "http://127.0.0.1:5000",
  });

  const fetchQRCode = useCallback(async () => {
    try {
      const response = await api.get("api/get_short_code", {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }, [token]);

  const verifyQRCode = useCallback(async (customerId: string, code:string) => {
    try {
      const response = await api.get(`api/verify_qrcode/${customerId}/${code}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return response.data.success;
    } catch (error) {
      console.error(error);
      return false
    }
  }, [token, api]);

  const fetchVouchers = useCallback(
    async (customerId: string) => {
      try {
        const response = await api.get(
          `/api/get_vouchers_customer/${customerId}`,
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );
        setCustomerVouchers(response.data.vouchers);
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const claimVoucher = useCallback(
    async (voucherId: string, customerId: string) => {
      try {
        const response = await api.post(
          `/api/claim_voucher`,
          {
            voucher_id: voucherId,
            customer_id: customerId,
          },
          {
            headers: {
              Authorization: `Bearer ${token}`,
            },
          }
        );

        // Refresh vouchers after a successful claim
        fetchVouchers(customerId);
        return response.data;
      } catch (err) {
        const error = err as ErrorResponse;
        if (error.response && error.response.data) {
          alert(error.response.data.vouchers);
        } else {
          alert("An error occurred while claiming voucher.");
        }
        throw error;
      }
    },
    [token, fetchVouchers]
  );

  const fetchVouchersForEatery = useCallback(
    async (eateryId: string) => {
      try {
        const response = await api.get(`/api/get_vouchers_eatery/${eateryId}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        setEateryVouchers(response.data.vouchers);
      } catch (error) {
        console.error(error);
      }
    },
    [token]
  );

  const addVoucher = useCallback(
    async (formData: AddVoucher) => {
      try {
        const response = await api.post(`/api/create_voucher`, formData, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });

        return response.data;
      } catch (error) {
        console.error(error);
        throw error;
      }
    },
    [token]
  );

  const deleteVoucher = useCallback(
    async (voucherId: string) => {
      try {
        await api.delete(`/api/delete_voucher/${voucherId}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        return true; // return the success status
      } catch (error) {
        console.error(error + " ASSS");
      }
    },
    [token]
  );

  const deleteCustomerVoucher = useCallback(
    async (voucherId: string, customerId: string) => {
      try {
        await api.delete(`/api/delete_customer_voucher/${voucherId}/${customerId}`, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });
        return true; // return the success status
      } catch (error) {
        console.error(error);
      }
    },
    [token, api]
  );

  return (
    <VoucherContext.Provider
      value={{
        customerVouchers,
        fetchVouchers,
        claimVoucher,
        fetchVouchersForEatery,
        eateryVouchers,
        fetchQRCode,
        addVoucher,
        deleteVoucher,
        verifyQRCode,
        deleteCustomerVoucher
      }}
    >
      {children}
    </VoucherContext.Provider>
  );
};