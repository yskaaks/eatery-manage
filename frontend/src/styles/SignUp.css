body {
  margin: 0;
  font-family: sans-serif;
}

.home {
  background-color: #FFF7EC;
  height: 100vh;
}

.address-autocomplete {
  width: 100%;
}

.signup-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  padding: 0px 10px 0px 10px;
  background-color: #FFE8C9;
}

.signup-title {
  display: flex;
  color: #333;
  margin-bottom: 1rem;
}

.signup-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-width: 400px;
}

.input-field,
.submit-button,
select {
  width: 100%;
  padding: 10px;
  border-radius: 20px;
  border: 1px solid #ccc;
  margin-bottom: 10px;
  outline: none;
  box-sizing: border-box;

}

.input-field:focus,
.submit-button:focus,
select:focus {
  border-color: #007BFF;
  box-shadow: 0 0 5px #007BFF;
}

.submit-button {
  cursor: pointer;
  background-color: #007BFF;
  color: #fff;
  border-color: #007BFF;
}

.submit-button:hover {
  background-color: #0056b3;
  border-color: #0056b3;
}

.user-type-select button.selected {
  background-color: #ccc;
}


.user-type-select {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 10px;
}

.user-type-select button {
    width: 100%; 
    margin-bottom: 10px;
    padding: 10px;
    border-radius: 20px;
    border: 1px solid #ccc;
    background-color: #fff;
    outline: none;
    box-sizing: border-box;
    cursor: pointer;
    transition: background-color 0.3s ease, border-color 0.3s ease;
}

.user-type-select button:focus {
  border-color: #007BFF;
  box-shadow: 0 0 5px #007BFF;
}

.user-type-select button.selected {
  background-color: #007BFF;
  color: #fff;
  border-color: #007BFF;
}

.title-link {
  cursor: pointer;
  color: #0056b3;
}

.signup-sub-title {
  display: flex;
  padding-bottom: 10px;
}

.forgot-password {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.reset-password {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.toggle-reset-password-container {
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
}

.reset-button {
  cursor: pointer;
  background-color: #E07893;
  color: #fff;
  border-color: #E07893;
  width: 90%;
  padding: 10px;
  border-radius: 20px;
  border: 1px solid;
  margin-bottom: 10px;
  outline: none;
  box-sizing: border-box;
}

.forgot-password-text {
  color: #007BFF
}

.forgot-password-container {
  width: 100%;
  height: 100%;
  background-color: #FFE8C9;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.forgot-password-sub-title {
  display: flex;
  margin-bottom: 10px;
  font-size: x-large;
}