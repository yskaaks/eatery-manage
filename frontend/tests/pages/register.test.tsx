import { rest } from 'msw';
import React from "react";
import { setupServer } from 'msw/node';
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import SignUp from '../../src/pages/SignUp'
import { AuthProvider } from '../../src/context/AuthContext';
import { expect } from 'chai';
import { BrowserRouter as Router} from 'react-router-dom'


// Here, replace '/auth/register' with your actual API endpoint
const server = setupServer(
  rest.post('/auth/register', (req, res, ctx) => {
    return res(ctx.json({ message: 'success' }));
  }),
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test('SignUp works', async () => {
  const { getByPlaceholderText, getByText } = render(
    <AuthProvider>
      <Router>
        <SignUp />
      </Router>
    </AuthProvider>
  );

  // Fill out the form
  fireEvent.change(getByPlaceholderText('Name'), { target: { value: 'test user' } });
  fireEvent.change(getByPlaceholderText('Email'), { target: { value: 'test@example1.com' } });
  fireEvent.change(getByPlaceholderText('Password'), { target: { value: 'password' } });
  fireEvent.click(getByText("I'm a Customer"));

  // Submit the form
  fireEvent.click(getByText('Sign Up'));

  // Wait for promises to resolve
  await waitFor(() => {
    expect(screen.findByText('success')).to.exist;
  });
});
