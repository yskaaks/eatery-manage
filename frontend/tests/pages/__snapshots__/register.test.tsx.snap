// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignIn changes the class when hovered 1`] = `
<div
  className="signup-container"
>
  <h2
    className="signup-title"
  >
    Create an account or log in
  </h2>
  <p
    className="signup-sub-title"
  >
    Create a new account below or log in
  </p>
  <form
    className="signup-form"
    onSubmit={[Function]}
  >
    <input
      className="input-field"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Name"
    />
    <input
      className="input-field"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <input
      className="input-field"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <div
      className="user-type-select"
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        I'm a Customer
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        I'm a Resturant Owner
      </button>
    </div>
    <button
      className="submit-button"
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
